blocks/hooks/autocompleters.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { clone } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { addFilter, applyFilters } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import defaultCompleters from '../autocompleters';
+
+/**
+ * Wrap the default Autocomplete component with one that
+ * supports a filter hook for customizing its list of autocompleters.
+ *
+ * Since they may be many Autocomplete instances at one time, this component
+ * applies the filter on demand, when the component is first focused after
+ * receiving a new list of completers.
+ *
+ * This function is exported directly for unit test.
+ *
+ * @param  {Function} Autocomplete Original component.
+ * @return {Function}              Wrapped component
+ */
+export function filterAutocomplete( Autocomplete ) {
+	return class FilteredAutocomplete extends Component {
+		constructor() {
+			super();
+
+			this.state = {
+				completers: [],
+				hasStaleCompleters: true,
+			};
+
+			this.saveParentRef = this.saveParentRef.bind( this );
+			this.onFocus = this.onFocus.bind( this );
+		}
+
+		componentWillReceiveProps( { completers: nextCompleters } ) {
+			if ( nextCompleters !== this.props.completers ) {
+				const hasFocus = this.parentNode.contains( document.activeElement );
+
+				if ( hasFocus ) {
+					this.updateFilteredCompleters( nextCompleters );
+				} else {
+					this.expireFilteredCompleters();
+				}
+			}
+		}
+
+		updateFilteredCompleters( nextCompleters ) {
```

Comment:
It can be also simplified to:
```js
updateFilteredCompleters( nextCompleters = defaultCompleters ) {
```
by using a default param value.

Reply:
Thanks, I'm not sure how I missed that :).  Based on one of your other comments, I've updated the PR to provide defaults via a filter instead, so this no longer applies.